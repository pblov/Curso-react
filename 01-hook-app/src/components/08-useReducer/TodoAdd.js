import React from 'react'
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {


    const [{description}, handleInputChange,reset] = useForm({
        description: ''
    });

    const handleSubmit = (e) => {

        e.preventDefault();

        if( description.trim().length <= 1){
            return;
        } 


        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }

        handleAddTodo(newTodo);
        reset();

    }

    return (
        <form action="" onSubmit={ handleSubmit }>
            <h3>Agregar TODO's</h3>
            <hr />
            <input className="form-control" 
                    type="text" 
                    name="description" 
                    placeholder="Aprender ..." 
                    autoComplete ="off" 
                    onChange = {handleInputChange}
                    value = {description}
            />
            <button type="submit"className="btn btn-outline-primary mt-3 w-100"> Agregar  TO-DO's</button>

        </form>
    )
}
