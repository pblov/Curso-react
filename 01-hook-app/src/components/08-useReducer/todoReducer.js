export const todoReducer = (state = [], action) => {

    switch ( action.type ) {
        case 'add':
                 return [...state, action.payload];
    
        case 'delete':
                 return state.filter( todo => todo.id !== action.payload ); //regresa todos los elementos del arreglo state [] que cumplan una condiciÃ³n.
        
        case 'toggle-old':
                 return state.map( todo => {

                    if (todo.id  === action.payload ){
                        return {
                            ...todo,
                            done: !todo.done
                        }
                    }else{
                        return todo;
                    }

                 })

            case 'toggle':
                 return state.map(todo => 
                    (todo.id === action.payload)
                    ? { ...todo, done:!todo.done}
                    : todo
                    )
        
        default:
            return state;
    }

}