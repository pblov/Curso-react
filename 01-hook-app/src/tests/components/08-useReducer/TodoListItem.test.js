import { shallow } from 'enzyme';
import React from 'react';
import { TodoListItem } from '../../../components/08-useReducer/TodoListItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('pruebas en /TodoListItem/', () => {
    
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();


    const wrapper = shallow(
        <TodoListItem
            todo = {demoTodos[0]}
            index = {0}
            handleDelete = { handleDelete }
            handleToggle = { handleToggle }
        
        />
    )

    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    })


})
