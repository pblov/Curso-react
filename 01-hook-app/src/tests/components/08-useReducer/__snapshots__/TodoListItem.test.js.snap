// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en /TodoListItem/ debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    <b>
      NaN
      .-
    </b>
     
    Aprender react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
